File: Software Technologies/Exam SoftUni Lab 10.Jan.2019/Java/softunilabs/src/main/java/softunilabs/controllers/LabController.java
package softunilabs.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import softunilabs.bindingModels.LabBindingModel;
import softunilabs.entities.Lab;
import softunilabs.repositories.LabRepository;

import java.util.List;

@Controller
public class LabController {
    private final LabRepository labRepository;

    @Autowired
    public LabController(LabRepository labRepository) {
        this.labRepository = labRepository;
    }

    @GetMapping("/")
    public String index(Model model) {
        //TODO: Implement me

        List<Lab> labs = labRepository.findAll();
        model.addAttribute("labs", labs);
        model.addAttribute("view", "lab/index");

        return "base-layout";

    }

    @GetMapping("/create")
    public String create(Model model) {
        //TODO: Implement me

        model.addAttribute("view", "lab/create");
        return "base-layout";
    }

    @PostMapping("/create")
    public String create(LabBindingModel bindingModel) {
        //TODO: Implement me

        Lab lab = new Lab();
        lab.setName(bindingModel.getName());
        lab.setCapacity(bindingModel.getCapacity());
        lab.setStatus(bindingModel.getStatus());

        labRepository.saveAndFlush(lab);

        return "redirect:/";
    }

    @GetMapping("/edit/{id}")
    public String edit(Model model,
                       @PathVariable(value = "id") Integer id) {
        //TODO: Implement me

        Lab lab = labRepository.findOne(id);
        model.addAttribute("lab", lab);
        model.addAttribute("view", "lab/edit");
        return "base-layout";

    }

    @PostMapping("/edit/{id}")
    public String edit(LabBindingModel bindingModel,
                       @PathVariable(value = "id") Integer id) {
        //TODO: Implement me

        Lab lab = labRepository.findOne(id);
        lab.setName(bindingModel.getName());
        lab.setCapacity(bindingModel.getCapacity());
        lab.setStatus(bindingModel.getStatus());

        labRepository.saveAndFlush(lab);

        return "redirect:/";
    }

    @GetMapping("/delete/{id}")
    public String delete(Model model,
                         @PathVariable(value = "id") Integer id) {
        //TODO: Implement me

        Lab lab = labRepository.findOne(id);
        model.addAttribute("lab", lab);
        model.addAttribute("view", "lab/delete");

        return "base-layout";
    }

    @PostMapping("/delete/{id}")
    public String delete(@PathVariable(value = "id") Integer id) {
        //TODO: Implement me

        labRepository.delete(id);
        return "redirect:/";
    }
}

File: Software Technologies/Exam SoftUni Lab 10.Jan.2019/CSharp/SoftUniLabs/SoftUniLabs/Controllers/LabController.cs
using System.Collections.Generic;
using System.Linq;
using Microsoft.AspNetCore.Mvc;
using SoftUniLabs.Data;
using SoftUniLabs.Models;

namespace SoftUniLabs.Controllers
{
    public class LabController : Controller
    {
        private readonly LabDbContext context;

        public LabController(LabDbContext context)
        {
            this.context = context;
        }

        [HttpGet]
        [Route("/")]
        public IActionResult Index()
        {
            //TODO: Implement me

            var labs = context.Labs.ToList();
            return View(labs);
            
        }

        [HttpGet]
        [Route("/create")]
        public IActionResult Create()
        {
            //TODO: Implement me
            return View();
        }

        [HttpPost]
        [Route("/create")]
        public IActionResult Create(Lab lab)
        {
            //TODO: Implement me

            context.Labs.Add(lab);
            context.SaveChanges();
            return RedirectToAction("Index");
        }

        [HttpGet]
        [Route("/edit/{id}")]
        public IActionResult Edit(int id)
        {
            //TODO: Implement me

            Lab lab = context
                            .Labs
                            .Where(p => p.Id == id)
                            .FirstOrDefault();

            return View(lab);
        }

        [HttpPost]
        [Route("/edit/{id}")]
        public IActionResult Edit(Lab lab)
        {
            //TODO: Implement me

            context.Labs.Update(lab);
            context.SaveChanges();

            return RedirectToAction("Index");
            
        }

        [HttpGet]
        [Route("/delete/{id}")]
        public IActionResult Delete(int id)
        {
            //TODO: Implement me

            Lab lab = context
                .Labs
                .Where(p => p.Id == id)
                .FirstOrDefault();

            return View(lab);
        }

        [HttpPost]
        [Route("/delete/{id}")]
        public IActionResult Delete(Lab lab)
        {
            //TODO: Implement me

            context.Labs.Remove(lab);
            context.SaveChanges();

            return RedirectToAction("Index");
        }
    }
}
